src/app/query/bitcoin/fees/fee-estimates.hooks.ts

```diff
@@ -1,9 +1,55 @@
+import BigNumber from 'bignumber.js';
+
 import { logger } from '@shared/logger';
 
-import { useGetBitcoinFeeEstimatesQuery } from './fee-estimates.query';
+import { calculateMeanAverage } from '@app/common/calculate-averages';
+
+import {
+  useGetBitcoinAllFeeEstimatesQuery,
+  useGetBitcoinMempoolApiFeeEstimatesQuery,
+} from './fee-estimates.query';
 
 export function useBitcoinFeeRate() {
-  return useGetBitcoinFeeEstimatesQuery({
+  return useGetBitcoinMempoolApiFeeEstimatesQuery({
     onError: err => logger.error('Error getting bitcoin fee estimates', { err }),
   });
 }
+
+export function useAverageBitcoinFeeRate() {
+  const { data: avgApiFeeRates } = useGetBitcoinAllFeeEstimatesQuery({
+    onError: err => logger.error('Error getting all apis bitcoin fee estimates', { err }),
+    select: resp => {
+      if (resp[0].status === 'rejected' && resp[1].status === 'rejected') {
+        return null;
+      }
+
+      let mempoolApiFeeRates = null;
+      if (resp[0].status === 'fulfilled') {
+        mempoolApiFeeRates = resp[0].value;
+      }
+
+      let earnApiFeeRates = null;
+      if (resp[1].status === 'fulfilled') {
+        earnApiFeeRates = resp[1].value;
+      }
+
+      // zero values for cases when one api is down
+      return {
+        fastestFee: calculateMeanAverage([
+          new BigNumber(mempoolApiFeeRates?.fastestFee ?? 0),
+          new BigNumber(earnApiFeeRates?.fastestFee ?? 0),
```

Comment:
Not sure we want to default to `0`, because it'll halve the fee if one isn't resolved. Maybe do a filter?

I would test with chrome dev tools to block 1 and see what happens.

Reply:
ah, yeah. exactly. I also added loading spinner and no fees state